examples/bq_file_load_benchmark/benchmark_tools/tables_processor.py

```diff
@@ -0,0 +1,165 @@
+        duplicate_benchmark_tables(bool): Boolean value to determine what to
+            do if a benchmark table already exists for a given file
+            combination. If True, TableProcessor knows to create another
+            benchmark table with the same combination to increase the
+            number of results for accuracy. If not, TablesProcessor knows
+            to only create a benchmark table for a given combination if one
+            has not yet been created.
+        file_params(dict): Dictionary containing each file parameter and
+            its possible values.
+        bq_logs_dataset(str): Name of dataset hold BQ logs table.
+
+    """
+
+    def __init__(
+            self,
+            bq_project,
+            gcs_project,
+            staging_project,
+            staging_dataset_id,
+            dataset_id,
+            bucket_name,
+            results_table_name,
+            results_table_dataset_id,
+            duplicate_benchmark_tables,
+            file_params,
+            bq_logs_dataset,
+    ):
+        self.bq_project = bq_project
+        self.gcs_project = gcs_project
+        self.staging_project = staging_project
+        self.staging_dataset_id = staging_dataset_id
+        self.dataset_id = dataset_id
+        self.bucket_name = bucket_name
+        self.bucket_util = bucket_util.BucketUtil(
+            bucket_name=self.bucket_name,
+            project_id=self.gcs_project,
+            file_params=file_params,
+        )
+        self.results_table_name = results_table_name
+        self.results_table_dataset_id = results_table_dataset_id
+        self.duplicate_benchmark_tables = duplicate_benchmark_tables
+        self.bq_logs_dataset = bq_logs_dataset
+
+    def gather_files_with_benchmark_tables(self):
+        """Generates file combinations that already have benchmark tables.
+
+        Creates a set of files that already have been loaded to create
+        benchmark tables. Generates list by querying the job.sourceURI field
+        from the results table.
+        Returns:
+            Set of file names that already have been loaded to create
+            benchmark tables.
+        """
+        query = (
+            'SELECT job.sourceURI FROM `{0:s}.{1:s}.{2:s}` '.format(
+                self.bq_project,
+                self.results_table_dataset_id,
+                self.results_table_name,
+            )
+        )
+        query_job = bigquery.Client().query(
+            query,
+            location='US',
+        )
+        files_with_benchmark_tables = set()
+        for row in query_job:
+            if self.bucket_name in row['sourceURI']:
+                uri = row['sourceURI'].split('gs://{0:s}/'.format(
+                    self.bucket_name
+                ))[1]
+                file_name = uri.split('/*')[0]
+                files_with_benchmark_tables.add(file_name)
+        return files_with_benchmark_tables
+
+    def create_benchmark_tables(self):
```

Comment:
INFO:root:Done discovering 7 existing files.
INFO:root:Processing  benchmark table for fileType=csv/compression=none/numColumns=10/columnTypes=100_STRING/numFiles=100/tableSize=2147MB
INFO:root:Processing  benchmark table for fileType=csv/compression=none/numColumns=10/columnTypes=100_STRING/numFiles=1000/tableSize=2147MB
INFO:root:Processing  benchmark table for fileType=csv/compression=none/numColumns=10/columnTypes=100_STRING/numFiles=1/tableSize=2147MB
INFO:root:Processing  benchmark table for fileType=csv/compression=gzip/numColumns=10/columnTypes=100_STRING/numFiles=1/tableSize=2147MB
INFO:root:Processing  benchmark table for fileType=csv/compression=gzip/numColumns=10/columnTypes=100_STRING/numFiles=1000/tableSize=2147MB
INFO:root:Processing  benchmark table for fileType=csv/compression=none/numColumns=10/columnTypes=100_STRING/numFiles=10000/tableSize=2147MB
INFO:root:Processing  benchmark table for fileType=csv/compression=gzip/numColumns=10/columnTypes=100_STRING/numFiles=100/tableSize=2147MB
INFO:root:Created new table: 1564792420
INFO:root:Started load job 29a123d2-6ad0-4351-855e-a7d265d95100 for table 1564792420.